src/utils/private.ts

```diff
@@ -0,0 +1,157 @@
+        format: "raw",
+      },
+    });
+    return data as unknown as string;
+  } catch (error: any) {
+    return undefined;
+  }
+};
+
+interface WideLabel {
+  name: string;
+  weight: number;
+  value?: number | undefined;
+  target: string;
+}
+
+interface WideConfig {
+  baseMultiplier?: number;
+  timeLabels?: WideLabel[];
+  priorityLabels?: WideLabel[];
+  autoPayMode?: boolean;
+  analyticsMode?: boolean;
+}
+
+interface WideRepoConfig extends WideConfig {}
+
+interface WideOrgConfig extends WideConfig {
+  PSK?: string;
+}
+
+interface DataConfig {
+  privateKey: string;
+  baseMultiplier: number;
+  timeLabels: WideLabel[];
+  priorityLabels: WideLabel[];
+  autoPayMode: boolean;
+  analyticsMode: boolean;
+}
+
+export const parseYAML = async (data: any): Promise<any | undefined> => {
+  try {
+    const parsedData = await YAML.parse(data);
+    if (parsedData !== null) {
+      return parsedData;
+    } else {
+      return undefined;
+    }
+  } catch (error) {
+    return undefined;
+  }
+};
+
+export const getPrivateKey = async (cipherText: string): Promise<string | undefined> => {
+  try {
+    await _sodium.ready;
+    const sodium = _sodium;
+    const {
+      sodium: { publicKey, privateKey },
+    } = getBotConfig();
+
+    if (publicKey === "" || privateKey === "") {
+      return undefined;
+    }
+
+    const binPub = sodium.from_base64(publicKey, sodium.base64_variants.URLSAFE_NO_PADDING);
+    const binPriv = sodium.from_base64(privateKey, sodium.base64_variants.URLSAFE_NO_PADDING);
+    const binCipher = sodium.from_base64(cipherText, sodium.base64_variants.URLSAFE_NO_PADDING);
+
+    let walletPrivateKey: string | undefined = sodium.crypto_box_seal_open(binCipher, binPub, binPriv, "text");
+    walletPrivateKey = walletPrivateKey.startsWith(KEY_PREFIX) ? walletPrivateKey.replace(KEY_PREFIX, "") : undefined;
+    return walletPrivateKey;
+  } catch (error: any) {
+    return undefined;
+  }
+};
+
+export const getScalarKey = async (X25519_PRIVATE_KEY: string | undefined): Promise<string | undefined> => {
+  try {
+    if (X25519_PRIVATE_KEY !== undefined) {
+      await _sodium.ready;
+      const sodium = _sodium;
+
+      const binPriv = sodium.from_base64(X25519_PRIVATE_KEY, sodium.base64_variants.URLSAFE_NO_PADDING);
+      const scalerPub = sodium.crypto_scalarmult_base(binPriv, "base64");
+      return scalerPub;
+    }
+    return undefined;
+  } catch (error: any) {
+    return undefined;
+  }
+};
+
+export const getWideConfig = async (context: Context): Promise<DataConfig> => {
+  const orgConfig = await getConfigSuperset(context, "org");
+  const repoConfig = await getConfigSuperset(context, "repo");
+
+  const parsedOrg: WideOrgConfig | undefined = await parseYAML(orgConfig);
+  const parsedRepo: WideRepoConfig | undefined = await parseYAML(repoConfig);
+  const PSK = parsedOrg && parsedOrg[KEY_NAME] ? await getPrivateKey(parsedOrg[KEY_NAME]) : undefined;
+
+  const configData: DataConfig = {
```

Comment:
Can't we merge configs together like `Object.assign(parsedOrg, parsedRepo);` ?

Reply:
> Can't we merge configs together like `Object.assign(parsedOrg, parsedRepo);` ?

There are additional checks that needs to be passed, Object.assign is prone to errors.

Reply:
My point is that (at least for me) this is really hard to read:
```
parsedRepo && parsedRepo["baseMultiplier"] && !Number.isNaN(Number(parsedRepo["baseMultiplier"]))
        ? Number(parsedRepo["baseMultiplier"])
        : parsedOrg && parsedOrg["baseMultiplier"] && !Number.isNaN(Number(parsedOrg["baseMultiplier"]))
        ? Number(parsedOrg["baseMultiplier"])
        : Number(DefaultPriceConfig["baseMultiplier"]),
``` 

So I'm trying to figure out the way to make it cleaner, perhaps by deep merging 2 yaml configs

Reply:
you need to tinker around ternary to get used to it.

Reply:
We shouldn't use ternary or any shorthand. 

Code readability and maintenance outweighs the benefits of saving a few keystrokes.

Reply:
refactored